sources/Outputs/src/Par.cpp

```diff
@@ -128,12 +127,14 @@ Par::writeConstantSets(unsigned int nb_generators) {
 
 boost::shared_ptr<parameters::ParametersSet>
 Par::writeGenerator(const algo::GeneratorDefinition& def, const std::string& basename) {
-  auto set = parameters::ParametersSetFactory::newInstance(def.id);
-
   if (def.model == algo::GeneratorDefinition::ModelType::SIGNALN || def.model == algo::GeneratorDefinition::ModelType::WITH_IMPEDANCE_SIGNALN) {
     // already processed by constant
     return nullptr;
   }
+  std::size_t hashId = constants::hash(def.id);
+  std::string hashIdStr = std::to_string(hashId);
+  auto set = parameters::ParametersSetFactory::newInstance(hashIdStr);
+  //change def.id to the hash
```

Comment:
This coment is irrelevant for final code. change it something like "use the hash id in exported files to prevent use of non-ascii characters"